### PR TITLE
chore: fix build (vue, 3, vue-cli, latest, yarn, ts, 18)

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

One of our build system tests has high failure rate - [build (vue, 3, vue-cli, latest, yarn, ts, 18)](https://github.com/aws-amplify/amplify-ui/actions/runs/5650145379/job/15305956262#logs). Digging into the test, the root cause is the dependency installation was interrupted once in a while by a prompt asking if switching to another registry when there was a slow connection and thus failed the following steps.

```
Your connection to the default yarn registry seems to be slow.
  Use https://registry.npmmirror.com/ for faster installation? (Y/n)
```

Answering a `Y` to the prompt whenever it comes up can fix the issue:
```bash
echo 'Y' | vue create --preset ../templates/components/vue/preset-${FRAMEWORK_VERSION}.json $MEGA_APP_NAME
```

Also, fix 2 commands which are supposed to use `yarn` to install dependencies but we forgot to pass the flag.

Before:
- [build (react, latest, cra, latest, yarn, 1, ts, 18)](https://github.com/aws-amplify/amplify-ui/actions/runs/5650264212/job/15306310361#step:4:1002)
- [build (vue, 3, vue-cli, latest, yarn, ts, 18)](https://github.com/aws-amplify/amplify-ui/actions/runs/5650264212/job/15306311312#step:4:101)

After:
- [build (react, latest, cra, latest, yarn, 1, ts, 18)](https://github.com/zchenwei/amplify-ui/actions/runs/5650328407/job/15306960452#step:4:1005)
- [build (vue, 3, vue-cli, latest, yarn, ts, 18)](https://github.com/zchenwei/amplify-ui/actions/runs/5650328407/job/15306486043#step:4:115)

#### Description of how you validated changes

[build (vue, 3, vue-cli, latest, yarn, ts, 18](https://github.com/zchenwei/amplify-ui/actions/runs/5650638753/job/15307471987#step:4:48)

[Overall POC run](https://github.com/zchenwei/amplify-ui/actions/runs/5650638753)

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-system-tests/scripts/mega-app-install.sh [hunk 1]
```diff
@@ -88,13 +88,12 @@ cd ./mega-apps/${MEGA_APP_NAME}
...
 if [ "$FRAMEWORK" == 'react' ]; then
     # add react-dom
-    echo "DEPENDENCIES='\$DEPENDENCIES react-dom@$FRAMEWORK_VERSION @aws-amplify/ui-react-storage @aws-amplify/ui-react-geo @aws-amplify/ui-react-notifications'"
+    echo "DEPENDENCIES='$DEPENDENCIES react-dom@$FRAMEWORK_VERSION @aws-amplify/ui-react-storage @aws-amplify/ui-react-geo @aws-amplify/ui-react-notifications'"
     DEPENDENCIES="$DEPENDENCIES react-dom@$FRAMEWORK_VERSION @aws-amplify/ui-react-storage @aws-amplify/ui-react-geo @aws-amplify/ui-react-notifications"
 
     if [[ "$BUILD_TOOL" == 'cra' && "$LANGUAGE" == 'ts' ]]; then
         DEP_TYPES="@types/react@$FRAMEWORK_VERSION @types/react-dom@$FRAMEWORK_VERSION"
-        echo "yarn add $DEP_TYPES"
-        yarn add $DEP_TYPES
```
Comment (zchenwei) [inline build-system-tests/scripts/mega-app-install.sh:97]: No need to install here.  Redundant step

File: build-system-tests/scripts/mega-app-create-app.sh [hunk 2]
```diff
@@ -109,7 +109,7 @@ if [[ "$FRAMEWORK" == 'vue' ]]; then
     npm install -g @vue/cli@${BUILD_TOOL_VERSION}
     if [ "$BUILD_TOOL" == 'vue-cli' ]; then
         echo "vue create --preset ../templates/components/vue/preset-${FRAMEWORK_VERSION}.json $MEGA_APP_NAME"
-        vue create --preset ../templates/components/vue/preset-${FRAMEWORK_VERSION}.json $MEGA_APP_NAME
+        printf 'Y\n' | vue create --preset ../templates/components/vue/preset-${FRAMEWORK_VERSION}.json $MEGA_APP_NAME
```
Comment (ioanabrooks) [inline build-system-tests/scripts/mega-app-create-app.sh:112]: I think iirc echo prints the end of line by default.

```suggestion
        echo 'Y' | vue create --preset ../templates/components/vue/preset-${FRAMEWORK_VERSION}.json $MEGA_APP_NAME
```

**Review by ioanabrooks — COMMENTED** [review]
Thanks for doing this!

**Review by ioanabrooks — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request merged**